cmd/era/main.go

```diff
@@ -0,0 +1,334 @@
+		Name:      "verify",
+		ArgsUsage: "<expected>",
+		Usage:     "verifies each era against expected accumulator root",
+		Action:    verify,
+	}
+)
+
+func init() {
+	app.Commands = []*cli.Command{
+		blockCommand,
+		infoCommand,
+		verifyCommand,
+	}
+	app.Flags = []cli.Flag{
+		dirFlag,
+		networkFlag,
+		eraSizeFlag,
+	}
+}
+
+func main() {
+	if err := app.Run(os.Args); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+// block prints the specified block from an Era store.
+func block(ctx *cli.Context) error {
+	num, err := strconv.ParseUint(ctx.Args().First(), 10, 64)
+	if err != nil {
+		return fmt.Errorf("invalid block number: %w", err)
+	}
+
+	f, err := open(ctx, num/uint64(ctx.Int(eraSizeFlag.Name)))
+	if err != nil {
+		return fmt.Errorf("error opening era: %w", err)
+	}
+	r := era.NewReader(f)
+
+	// Read block with number.
+	block, err := r.ReadBlock(num)
+	if err != nil {
+		return fmt.Errorf("error reading era: %w", err)
+	}
+
+	// Convert block to JSON and print.
+	val, err := ethapi.RPCMarshalBlock(block, ctx.Bool(txsFlag.Name), ctx.Bool(txsFlag.Name), params.MainnetChainConfig)
+	if err != nil {
+		return fmt.Errorf("error marshaling json: %w", err)
+	}
+	b, err := json.MarshalIndent(val, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error marshaling json: %w", err)
+	}
+	fmt.Println(string(b))
+	return nil
+}
+
+// info prints some high-level information about the Era file.
+func info(ctx *cli.Context) error {
+	epoch, err := strconv.ParseUint(ctx.Args().First(), 10, 64)
+	if err != nil {
+		return fmt.Errorf("invalid block number: %w", err)
+	}
+	f, err := open(ctx, epoch)
+	if err != nil {
+		return err
+	}
+	r := era.NewReader(f)
+
+	acc, err := r.Accumulator()
+	if err != nil {
+		return fmt.Errorf("error reading accumulator: %w", err)
+	}
+	td, err := r.InitialTD()
+	if err != nil {
+		return fmt.Errorf("error reading total difficulty: %w", err)
+	}
+	start, err := r.Start()
+	if err != nil {
+		return fmt.Errorf("error reading start block: %w", err)
+	}
+	count, err := r.Count()
+	if err != nil {
+		return fmt.Errorf("error reading count: %w", err)
+	}
+	info := struct {
+		Accumulator     common.Hash `json:"accumulator"`
+		TotalDifficulty *big.Int    `json:"totalDifficulty"`
+		StartBlock      uint64      `json:"startBlock"`
+		Count           uint64      `json:"count"`
+	}{
+		acc, td, start, count,
+	}
+	b, _ := json.MarshalIndent(info, "", "  ")
+	fmt.Println(string(b))
+	return nil
+}
+
+// open opens an era1 file at a certain epoch.
+func open(ctx *cli.Context, epoch uint64) (*os.File, error) {
+	var (
+		dir     = ctx.String(dirFlag.Name)
+		network = ctx.String(networkFlag.Name)
+	)
+	entries, err := era.ReadDir(dir, network)
+	if err != nil {
+		return nil, fmt.Errorf("error reading era dir: %w", err)
+	}
+	if epoch >= uint64(len(entries)) {
+		return nil, fmt.Errorf("epoch out-of-bounds: have %d, want %d", len(entries), epoch)
+	}
+	return os.Open(path.Join(dir, entries[epoch]))
```

Comment:
This only works if the era files are continuous right? I think if I have an `aaa.era`, or I'm missing a file it would open the wrong file

Reply:
Yeah `era.ReadDir` only returns non-error if the eras are sequential... not ideal, but I think solving this now is unnecessarily complicated when it isn't currently useful to have gaps in eras.

Reply:
You could return a map instead of an array with the filenames, but kinda agree that its unnecessary if we assume gaps in eras are impossible